File: frontend/src/types/discovery.ts
import type { Agent } from "./agent";

export type SearchMode = "keyword" | "semantic" | "capability" | "intent";

export interface SearchQuery {
  query: string;
  mode: SearchMode;
  category?: string;
  tags: string[];
  max_latency_ms?: number;
  max_credits?: number;
  input_modes: string[];
  output_modes: string[];
  min_reputation: number;
  limit: number;
}

export interface AgentMatch {
  agent: Agent;
  relevance_score: number;
  match_reason: string;
}

export interface DiscoveryResponse {
  matches: AgentMatch[];
  total_candidates: number;
  query_time_ms: number;
}

File: src/schemas/discovery.py
from enum import Enum
from typing import Optional

from pydantic import BaseModel

from src.schemas.agent import AgentResponse


class SearchMode(str, Enum):
    keyword = "keyword"
    semantic = "semantic"
    capability = "capability"
    intent = "intent"


class SearchQuery(BaseModel):
    query: str
    mode: SearchMode = SearchMode.keyword
    category: Optional[str] = None
    tags: list[str] = []
    max_latency_ms: Optional[int] = None
    max_credits: Optional[float] = None
    input_modes: list[str] = []
    output_modes: list[str] = []
    min_reputation: float = 0
    limit: int = 10


class AgentMatch(BaseModel):
    agent: AgentResponse
    relevance_score: float
    match_reason: str


class DiscoveryResponse(BaseModel):
    matches: list[AgentMatch]
    total_candidates: int
    query_time_ms: float
    hint: str | None = None
